internal/cmd/test_load.go

```diff
@@ -106,15 +110,32 @@ func loadLocalTest(gs *state.GlobalState, cmd *cobra.Command, args []string) (*l
 		preInitState:   state,
 	}
 
-	gs.Logger.Debugf("Initializing k6 runner for '%s' (%s)...", sourceRootPath, resolvedPath)
-	if err := test.initializeFirstRunner(gs); err != nil {
-		return nil, fmt.Errorf("could not initialize '%s': %w", sourceRootPath, err)
+	if err := test.prepareFirstRunner(gs); err != nil {
+		return test, fmt.Errorf("could not initialize '%s': %w", sourceRootPath, err)
 	}
-	gs.Logger.Debug("Runner successfully initialized!")
+
 	return test, nil
 }
 
-func (lt *loadedTest) initializeFirstRunner(gs *state.GlobalState) error {
+func loadLocalTest(gs *state.GlobalState, cmd *cobra.Command, args []string) (*loadedTest, error) {
+	test, err := loadLocalTestWithoutRunner(gs, cmd, args)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := test.continueInitialization(gs); err != nil {
+		return nil, fmt.Errorf("could not initialize '%s': %w", test.sourceRootPath, err)
+	}
+
+	return test, nil
+}
+
+//nolint:funlen
```

Comment:
I feel we can move out into a dedicated package some core parts for the resolution's operations. I'd be cool if we could improve them a bit in a follow-up pull request.

Reply:
After #5427 I will try to figure out some refactor. But now I do not think is the time to do it. 

Maybe as part of work around  #3259 as that also has to deal with some parts of this.